pkg/virtctl/vm/expand.go

```diff
@@ -0,0 +1,152 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package vm
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	"sigs.k8s.io/yaml"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/templates"
+)
+
+func NewExpandCommand(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "expand (VM)",
+		Short:   "Return the VirtualMachine object with expanded instancetype and preference.",
+		Example: usageExpand(),
+		Args:    cobra.MatchAll(cobra.ExactArgs(0), expandArgs()),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c := Command{command: COMMAND_EXPAND, clientConfig: clientConfig, cmd: cmd}
+			return c.expandRun(args)
+		},
+	}
+	cmd.Flags().StringVar(&vmName, vmArg, "", "Specify VirtualMachine name that should be expanded. Mutually exclusive with \"--file\" flag.")
+	cmd.Flags().StringVarP(&filePath, filePathArg, filePathArgShort, "", "If present, the Virtual Machine spec in provided file will be expanded. Mutually exclusive with \"--vm\" flag.")
+	cmd.Flags().StringVarP(&outputFormat, outputFormatArg, outputFormatArgShort, YAML, "Specify a format that will be used to display output.")
+	cmd.MarkFlagsMutuallyExclusive(filePathArg, vmArg)
+	cmd.SetUsageTemplate(templates.UsageTemplate())
+	return cmd
+}
+
+func (o *Command) expandRun(args []string) error {
+	vmOptions, err := BeforeRun(args, o.clientConfig)
+	if err != nil {
+		return err
+	}
+
+	return expandVirtualMachine(vmOptions.namespace, vmOptions.virtClient, o)
+}
+
+func usageExpand() string {
+	return `  #Expand a virtual machine called 'myvm'.
+  {{ProgramName}} expand --vm myvm
+  
+  # Expand a virtual machine from file called myvm.yaml.
+  {{ProgramName}} expand --file myvm.yaml
+
+  # Expand a virtual machine called myvm and display output in json format.
+  {{ProgramName}} expand --vm myvm --output json
+  `
+}
+
+func expandArgs() cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if filePath == "" && vmName == "" {
+			return fmt.Errorf("error invalid arguments - VirtualMachine name or file must be provided")
+		}
+
+		if outputFormat != YAML && outputFormat != JSON {
+			return fmt.Errorf("error not supported output format defined: %s", outputFormat)
+		}
+
+		return nil
+	}
+}
+
+func readVMFromFile(filePath string) (*v1.VirtualMachine, error) {
+	vm := &v1.VirtualMachine{}
+
+	readFile, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading file %+w", err)
+	}
+
+	err = yaml.Unmarshal(readFile, vm)
```

Comment:
Might change this to use `yaml.NewYAMLOrJSONDecoder` in a follow up? We also support both as output format.